### DISCUSSION TITLE
How to specify filename with client-zip

### DISCUSSION THREAD

**Discussion by chris-rudmin** [Touffy/dl-stream#3] [Q&A] [answered]
I'd like to pass the readable stream to client-zip while providing a filename (different than the fetch response filename).
Is it possible to do this with the current implementation?

**Comment by Touffy** [comment, marked as answer]
Yes. You need to transform the stream of Responses into a stream of objects with a `name` property (your custom filename) and an `input` property (the Response) and optionally `lastModified` [as specified here](https://github.com/Touffy/client-zip#usage). You can do that easily with an async generator (because DownloadStreams are AsyncIterable) :
```js
const dlstream = new DownloadStream(yourInputURLsOrRequests)
async function *namedInput() {
  for await (const response of dlstream) yield {
    name: "your custom name.ext",
    input: response,
    lastModified: response.headers.get('Last-Modified')
  }
}
const zipRersponse = downloadZip(namedInput())
```
Sometime in the near future, it will be possible to use a TransformStream (because all ReadableStreams will be AsyncIterable like a DownloadStream) or the `map` method inherited on the DownloadStream. And there are quite a few libraries providing `map` functions on AsyncIterables today, if you prefer.
